scenarios/aws/microVMs/microvms/files/download-vm-images.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+import json
+import subprocess
+import sys
+from pathlib import Path
+
+
+def usage():
+    print(f"Usage: {sys.argv[0]} <data-file>")
+
+
+def main(data_file):
+    with open(data_file) as f:
+        download_data = json.load(f)
+
+    images_to_download = []
+
+    for image in download_data:
+        checksum_file = Path(image["checksum_path"])
+        if not check_integrity(checksum_file):
+            print(f"Integrity check failed for checksum file: {checksum_file}, downloading image")
+            images_to_download.append(image)
+
+    curl_args = [
+        "curl",
+        "--no-progress-meter",
+        "--fail",
+        "--show-error",
+        "--retry",
+        "3",
+        "--parallel",
```

Comment:
I think we are missing `--parallel-max` to prevent the issue @usamasaqib was talking about.

Reply:
Fixed, also added a comment about it (I didn't know that `curl` [defaults to 50](https://everything.curl.dev/cmdline/urls/parallel.html) for `--parallel-max`)